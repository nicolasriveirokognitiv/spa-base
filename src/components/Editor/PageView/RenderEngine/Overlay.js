import React from 'react';
import PropTypes from 'prop-types';
import { blockPropTypes } from 'Constants/propTypes';
import { getBlockByType, StaticRenderController } from 'Components/BlockCollection';
import { Alert } from 'antd';

function Overlay({ type, mode, block }) {
    // const localizedBlock = useLocalizedContent(block); TODO: This hook should get the translated content
    const Block = getBlockByType(type);

    if (Block) {
        if (mode === 'static') {
            return (
                <StaticRenderController block={block}>
                    <Block.Static />
                </StaticRenderController>
            );
        }
        return <Block.Dynamic />;
    }

    return <Alert showIcon message="Oops!" description="Block not found" type="error" data-testid="alert" />;
}

Overlay.defaultProps = { block: {} };

Overlay.propTypes = {
    block : blockPropTypes,
    type  : PropTypes.string.isRequired,
    mode  : PropTypes.string.isRequired,
};

export default Overlay;
