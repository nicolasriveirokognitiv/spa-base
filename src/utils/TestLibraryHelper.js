/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import initStore from 'State/store';

const store = initStore();

const renderWithRedux = children => render(
    <Provider store={store}>
        {children}
    </Provider>,
);

export * from '@testing-library/react';
export { renderWithRedux, render };
