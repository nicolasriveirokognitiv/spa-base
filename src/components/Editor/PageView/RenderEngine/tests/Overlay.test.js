import React from 'react';
import { render, screen } from 'Utils/TestLibraryHelper';
import Overlay from '../Overlay';

describe('Overlay', () => {
    let props;
    let container;

    beforeEach(() => {
        container = document.createElement('div');
        document.body.appendChild(container);
        props = {
            block: {
                category : 'Headers',
                config   : { title: 'Dummy' },
                data     : { description: 'This is a notification ', title: 'Notification Title' },
                design   : { size: 'Medium' },
                id       : 'xaF0kvS',
                type     : 'Notification',
            },
            type : 'Notification',
            mode : 'static',
        };
    });

    test('Should render Correctly', () => {
        expect(<Overlay {...props} />).toBeDefined();
    });

    test('Should match snapshot', async () => {
        const { asFragment } = render(<Overlay {...props} />);
        expect(asFragment()).toMatchSnapshot();
    });

    test('Should show an alert if type is no correct', () => {
        props.type = 'Foo';
        render(<Overlay {...props} />);
        expect(screen.getByTestId('alert')).toBeDefined();
    });

    test('Should show an alert if type is no correct', () => {
        props.mode = 'dynamic';
        render(<Overlay {...props} />);
        expect(screen.getByTestId('notification-dynamic')).toBeDefined();
    });
});

