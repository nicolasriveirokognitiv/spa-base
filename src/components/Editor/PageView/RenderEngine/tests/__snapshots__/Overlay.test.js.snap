// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overlay Should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="container-full"
  >
    <div
      class="styled-wrapper    notification-block"
    >
      <div
        class="notification-block-title"
      >
        Notification Title
      </div>
      <div
        class="notification-block-description"
      >
        This is a notification 
      </div>
    </div>
  </div>
</DocumentFragment>
`;
